developer_docs/friendly_error_system.md

```diff
@@ -116,10 +116,18 @@ line(0, 0, 100, 100, x3, Math.PI);
 * Global Error catching. It would be very helpful to catch the errors the browser is throwing to the console, so we can match them up with friendly comments. So far we've tried `window.onerror` and the following with no success. 
 
```

Comment:
Is this snippet below something that works or does not work? In the previous version of this text, this paragraph was explaining something that had been tried but does not work.

Reply:
Please add 2 space indentations for this code block below.

Reply:
@lmccart  I have done the indentation modification as directed. 

In the previous version of the code , 
```
var original = window.console;
      
["log", "warn", "error"].forEach(function(func){
window.console[func] = function(msg) {
    original[func].apply(original, arguments)
};
``` 

The newly defined function was referencing itself, causing the stack overflow.

This was corrected with help from @limzykenneth , 
making sure that the new functions do not reference themselves , by storing the references to the original function separately. 
```
const original = window.console;
const original_functions  = {
  log: original.log,
  warn:  original.warn,
  error: original.error
}

["log", "warn", "error"].forEach(function(func){
window.console[func] = function(msg) {
//do something with the msg caught by the wrapper function, then call the original function
original_functions[func].apply(original, arguments)
};
});
```

So the updated snippet does not cause the overflow and also catches the errors.

Also , I was not fully able to understand the use of catching global errors ,
but if we want  to replace the red error messages with better looking custom messages, 
we can replace the window.onerror method and finally return true , which may cause the code to run as if it did not detect the error, but also remove the 'unfriendly' error message :smile: 
Maybe to handle errors , we can wrap the window.onerror method with some other function, which returns true.

Thank you!

Reply:
Got it, thank you @Ajayneethikannan. Then can we remove this line above: 
"So far weve tried `window.onerror` and the following with no success."

And update this comment:
```js
//this snippet attempts to wrap window.console methods with a new function to modify their functionality"
```
to

```js
// this snippet wraps window.console methods with a new function to modify their functionality
// it is not currently implemented, but could be to give nicer formatted error messages
```

Then we should be good to merge. Thanks!

Reply:
I have done the mentioned changes ,  please check it ,
Thank you @lmccart  !